progress/README.md

```diff
@@ -0,0 +1,313 @@
+# Progress
+
+## Requests
+
+|                     Route                     |                                          Description                                          |
+| :-------------------------------------------: | :-------------------------------------------------------------------------------------------: |
+|     [POST /progresses](#post-progresses)      |                                 Creates a new progress entry.                                 |
+|      [GET /progresses](#get-progresses)       |                 Retrieves progress entries based on the provided parameters.                  |
+| [GET /progresses/range](#get-progressesrange) | Retrieves the progress records for a particular user or task within the specified date range. |
+
+## POST /progresses
+
+    Creates a new progress entry.
+
+- **Params**  
+  None
+- **Query**
+  None
+- **Body**  
+  Attributes:
+  -type (required, string): Specifies the type of the progress entry (e.g., "task", "user").
+  -taskId (optional, string): Task ID associated with the progress entry (applicable for task progress only).
+  -completed (required, string): Completed portion of the task or progress.
+  -planned (required, string): Planned portion of the task or progress.
+  -blockers (required, string): Blockers for the progress entry.
+  Note : Since this is a authenticated route we don't need to pass userId in the request body.
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Success Response:**
+  - **Code:** 201
+    - **Content:**
+      ```json
+      {
+        "message": "String",
+        "data": {
+          "type": "String",
+          "completed": "String",
+          "planned": "String",
+          "blockers": "String",
+          "userId": "String",
+          "taskId": "String (for task type)",
+          "createdAt": "Timestamp",
+          "date": "Timestamp",
+          "id": "String"
+        }
+      }
+      ```
+- **Error Response:**
+
+  - **Code:** 400
+
+    - **Content:**
+      ```json
+      {
+        "message": "Bad Request"
+      }
+      ```
+  - **Code:** 401
+    - **Content:**
+      ```json
+      {
+        "statusCode": 401,
+        "error": "Unauthorized",
+        "message": "Unauthenticated User."
+      }
+      ```
+  - **Code:** 409
+    - **Content:**
+      ```json
+      {
+        "message": "User Progress for the day has already been created"
+      }
+      ```
+  - **Code:** 500
+    - **Content:**
+      ```json
+      {
+        "message": "Internal Server Error"
+      }
+      ```
```

Comment:
what about the case of an invalid task id it would be a 404 right? can we have that response mentioned here?

Reply:
Mentioned 👍🏻